appserver/web/weld-integration/src/main/java/org/glassfish/weld/FilteringClassLoader.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2022 Eclipse Foundation and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.weld;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.Runtime.Version;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.jar.Attributes;
+import java.util.jar.Manifest;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.enumeration;
+import static java.util.stream.StreamSupport.stream;
+
+/**
+ * This classloader filters out extensions which are incompatible with current environment.
+ *
+ * @author David Matejcek
+ */
+class FilteringClassLoader extends ClassLoader {
```

Comment:
I would rename the classloader to something that describes better what it does. For example, `JdkFilteringClassLoader` or `JavaRuntimeFilteringClassLoader`.

Reply:
It would be too verbose (impl detail), rules can change later. But more probable is that we will kill this CL after moving to JDK17.